`npm build` fails due to lack of pushd

When I try to build the spec on Ubuntu 20, I get:
```
tnewsome@compy-linux:~/SiFive/riscv-v-spec$ npm run build

> riscv-v-spec@0.1.0 build /home/tnewsome/SiFive/riscv-v-spec
> mkdir -p public && pushd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && popd

sh: 1: pushd: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! riscv-v-spec@0.1.0 build: `mkdir -p public && pushd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && popd`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the riscv-v-spec@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tnewsome/.npm/_logs/2020-10-09T16_21_20_651Z-debug.log
```

[2020-10-09T16_21_20_651Z-debug.log](https://github.com/riscv/riscv-v-spec/files/5356237/2020-10-09T16_21_20_651Z-debug.log)

Presumably this is because my /bin/sh doesn't have pushd, and on some systems pushd does work because /bin/sh is really /bin/bash.

I had the same problem this morning.

On Fri, Oct 9, 2020, 12:31 Tim Newsome, <notifications@github.com> wrote:

> When I try to build the spec on Ubuntu 20, I get:
>
> tnewsome@compy-linux:~/SiFive/riscv-v-spec$ npm run build
>
> > riscv-v-spec@0.1.0 build /home/tnewsome/SiFive/riscv-v-spec
> > mkdir -p public && pushd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && popd
>
> sh: 1: pushd: not found
> npm ERR! code ELIFECYCLE
> npm ERR! syscall spawn
> npm ERR! file sh
> npm ERR! errno ENOENT
> npm ERR! riscv-v-spec@0.1.0 build: `mkdir -p public && pushd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && popd`
> npm ERR! spawn ENOENT
> npm ERR!
> npm ERR! Failed at the riscv-v-spec@0.1.0 build script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
>
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/tnewsome/.npm/_logs/2020-10-09T16_21_20_651Z-debug.log
>
> 2020-10-09T16_21_20_651Z-debug.log
> <https://github.com/riscv/riscv-v-spec/files/5356237/2020-10-09T16_21_20_651Z-debug.log>
>
> Presumably this is because my /bin/sh doesn't have pushd, and on some
> systems pushd does work because /bin/sh is really /bin/bash.
>
I copied the string out of the json and ran in the public directory to make
it work.

> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/riscv/riscv-v-spec/issues/584>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFAWIKOMCVC7BQ2IZDKNHJDSJ43EHANCNFSM4SKJEODA>
> .
>

--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Hello RISC-V\"",
-    "build": "mkdir -p public && pushd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && popd"
+    "build": "mkdir -p public && cd public && ../node_modules/.bin/datasheet -i ../v-spec.adoc -o riscv-v-spec.pdf && cd .."
   },
   "repository": {
     "type": "git",

